### PR TITLE
Users can now schedule their profile pictures

### PR DESCRIPTION
**Pull request by gabebw**
* Add `TwitterClient.updateTwitterPicture`, which takes in a Base64-encoded
  image and sends it to Twitter as `application/x-www-form-urlencoded`
* Add a file input to the form
* Unfortunately switch from an applicative form to a monadic form, because the
  type of the file input is `FileInfo` and we need to do complicated (monadic)
  processing to turn that into a Base64-encoded `Maybe Text`

### CODE REVIEW

File: Handler/Profile.hs [hunk 1]
```diff
@@ -27,17 +32,38 @@ getProfileR :: Handler Html
 getProfileR = do
     euser@(Entity userId user) <- prerequisites
     tomorrow <- CT.localTomorrow user
-    profileFormPost <- generateFormPost $ profileForm tomorrow userId
-    profilesTemplate euser profileFormPost
+    (widget, _) <- generateFormPost $ profileForm tomorrow userId
```
Comment (tonyd256) [inline Handler/Profile.hs]: as an alternative, you could use `fst` or `snd` for these tuples if you didn't want to have the parens and ignore.

Comment (gabebw) [inline Handler/Profile.hs]: So like `widget <- fst <$> generateFormPost $ profileForm tomorrow userId` ?

Comment (tonyd256) [inline Handler/Profile.hs]: yes

File: Handler/Profile.hs [hunk 2]
```diff
@@ -27,17 +32,38 @@ getProfileR :: Handler Html
...
+fileContents :: (Text, FileInfo) -> Handler L.ByteString
+fileContents (_, fi) = runResourceT $ (fileSource fi) $$ sinkLbs
+
+-- If there's a profile picture in the HTTP request, set the Profile's
+-- profilePicture field to the Base64-encoded contents of that file.
+--
+-- If there's no profile picture, don't do anything.
+withPossibleProfilePicture :: FormResult Profile -> Handler (FormResult Profile)
+withPossibleProfilePicture (FormSuccess profile) = do
+    (_, files) <- runRequestBody
```
Comment (tonyd256) [inline Handler/Profile.hs]: also `files <- snd <$> runRequestBody` here

File: Handler/Profile.hs [hunk 3]
```diff
@@ -27,17 +32,38 @@ getProfileR :: Handler Html
...
+        base64text = b2t . B64.encode . toStrict
+
+withPossibleProfilePicture result = return result
+
 postProfileR :: Handler Html
 postProfileR = do
     euser@(Entity userId user) <- prerequisites
     tomorrow <- CT.localTomorrow user
-    ((result, formWidget), formEnctype) <- runFormPost $ profileForm tomorrow userId
+    ((resultWithoutProfilePicture, formWidget), _) <- runFormPost $ profileForm tomorrow userId
```
Comment (tonyd256) [inline Handler/Profile.hs]: `fst` might be nice here too

**Pull request merged**